tests_integ/helpers/base_test.py

```diff
@@ -0,0 +1,345 @@
+
+    def create_and_verify_stack(self, file_name, parameters=[]):
+        """
+        Creates the Cloud Formation stack and verifies it against the expected
+        result
+
+        Parameters
+        ----------
+        file_name : string
+            Template file name
+        parameters : list
+            List of parameters
+        """
+        self.output_file_path = str(Path(self.output_dir, "cfn_" + file_name + ".yaml"))
+        self.expected_resource_path = str(Path(self.expected_dir, file_name + ".json"))
+        self.stack_name = STACK_NAME_PREFIX + file_name.replace("_", "-") + "-" + generate_suffix()
+
+        self._fill_template(file_name)
+        self.transform_template()
+        self.deploy_stack(parameters)
+        self.verify_stack()
+
+    def transform_template(self):
+        transform_template(self.sub_input_file_path, self.output_file_path)
+
+    def get_region(self):
+        return self.my_region
+
+    def get_s3_uri(self, file_name):
+        """
+        Returns the S3 URI of a resource file
+
+        Parameters
+        ----------
+        file_name : string
+            Resource file name
+        """
+        return FILE_TO_S3_URL_MAP[file_name]["url"]
+
+    def get_code_key_s3_uri(self, code_key):
+        """
+        Returns the S3 URI of a code key for template replacement
+
+        Parameters
+        ----------
+        code_key : string
+            Template code key
+        """
+        return FILE_TO_S3_URL_MAP[CODE_KEY_TO_FILE_MAP[code_key]]["url"]
+
+    def get_stack_resources(self, resource_type, stack_resources=None):
+        if not stack_resources:
+            stack_resources = self.stack_resources
+
+        resources = []
+        for res in stack_resources["StackResourceSummaries"]:
+            if res["ResourceType"] == resource_type:
+                resources.append(res)
+
+        return resources
+
+    def get_stack_output(self, output_key):
+        for output in self.stack_description["Stacks"][0]["Outputs"]:
+            if output["OutputKey"] == output_key:
+                return output
+        return None
+
+    def get_stack_tags(self, output_name):
+        resource_arn = self.get_stack_output(output_name)["OutputValue"]
+        return self.sfn_client.list_tags_for_resource(resourceArn=resource_arn)["tags"]
+
+    def get_stack_deployment_ids(self):
+        resources = self.get_stack_resources("AWS::ApiGateway::Deployment")
+        ids = []
+        for res in resources:
+            ids.append(res["LogicalResourceId"])
+
+        return ids
+
+    def get_stack_stages(self):
+        resources = self.get_stack_resources("AWS::ApiGateway::RestApi")
+
+        if not resources:
+            return []
+
+        return self.api_client.get_stages(restApiId=resources[0]["PhysicalResourceId"])["item"]
+
+    def get_stack_stages_v2(self):
+        resources = self.get_stack_resources("AWS::ApiGatewayV2::Api")
+
+        if not resources:
+            return None
```

Comment:
Why do we return `None` here but `[]` for v1 above?

Reply:
Good catch, changing to []